carma-messenger-core/cpp_message/src/MobilityRequest_Message.cpp

```diff
@@ -0,0 +1,538 @@
+                location_timestamp+=int(message->value.choice.TestMessage00.body.location.timestamp.buf[i])-'0';
+            }
+            location.timestamp=location_timestamp;
+
+            output.location=location;    
+
+            //strategy params
+            std::string strategy_params;
+            str_len=message->value.choice.TestMessage00.body.strategyParams.size;
+            for(size_t i=0;i<str_len;i++){
+                strategy_params +=message->value.choice.TestMessage00.body.strategyParams.buf[i];
+            }
+            if(strategy_params==STRING_DEFAULT){
+                strategy_params="";
+            }
+            output.strategy_params=strategy_params;   
+
+            //Trajectory
+            cav_msgs::Trajectory trajectory;
+            cav_msgs::LocationECEF trajectory_start;
+            long tmp_t=message->value.choice.TestMessage00.body.trajectoryStart->ecefX;
+            if(tmp_t>LOCATION_MAX || tmp_t<LOCATION_MIN){
+                ROS_WARN_STREAM("Location ecefX is out of range");
+                return boost::optional<cav_msgs::MobilityRequest>{};
+            }
+            trajectory_start.ecef_x=tmp_t;
+
+            tmp_t=message->value.choice.TestMessage00.body.trajectoryStart->ecefY;
+            if(tmp_t>LOCATION_MAX || tmp_t<LOCATION_MIN){
+                ROS_WARN_STREAM("Location ecefY is out of range");
+                return boost::optional<cav_msgs::MobilityRequest>{};
+            }
+            trajectory_start.ecef_y=tmp_t;
+
+            tmp_t=message->value.choice.TestMessage00.body.trajectoryStart->ecefZ;
+            if(tmp_t>LOCATION_MAX_Z || tmp_t<LOCATION_MIN_Z){
+                ROS_WARN_STREAM("Location ecefZ is out of range");
+                return boost::optional<cav_msgs::MobilityRequest>{};
+            }
+            trajectory_start.ecef_z=tmp_t;
+
+            //convert location timestamp from string in asn1 to uint64 for ros message
+            str_len=message->value.choice.TestMessage00.body.trajectoryStart->timestamp.size;
+            uint64_t trajectory_timestamp=0;
+            for(size_t i=0;i<str_len;i++){
+                trajectory_timestamp*=10;
+                trajectory_timestamp+=int(message->value.choice.TestMessage00.body.trajectoryStart->timestamp.buf[i])-'0';
+            }
+            trajectory_start.timestamp=trajectory_timestamp;
+
+            trajectory.location=trajectory_start;
+
+            //Trajectory-offset
+            int offset_count=message->value.choice.TestMessage00.body.trajectory->list.count;
+            
+            if(offset_count>MAX_POINTS_IN_MESSAGE){
+            ROS_WARN_STREAM("offset count greater than 60.");
+            return boost::optional<cav_msgs::MobilityRequest>{};
+            } 
+
+            for(size_t i=0;i<offset_count;i++){
+                cav_msgs::LocationOffsetECEF Offsets;
+                Offsets.offset_x=message->value.choice.TestMessage00.body.trajectory->list.array[i]->offsetX;
```

Comment:
This reads oddly to me, though I can understand what's going on. I understand your reasoning for using bracketless same line if-statements as this would get pretty unwieldy without them. I'm not sure if there's a way to improve clarity of what's happening here, though. One thing to consider might be a ternary operator:

`Offsets.offset_y= (...array[i]->offSetY<OFFSET_MIN || ...array[i]->offSetY>OFFSET_MAX ? OFFSET_UNAVAILABLE : ...array[i]->offSetY`

If you pull out a reference to offSetX,Y,Z at the top of the loop:

```
long offset_y = message->value.choice.TestMessage00.body.trajectory->list.array[i]->offsetY
long offset_x = message->value.choice.TestMessage00.body.trajectory->list.array[i]->offsetX
...
```
Then maybe that's better for readability? I'm not sure. Worth putting a few minutes into to see if this section can be made easier to read/follow, even if the answer for now might just be adding whitespace/comments to make it flow better.